Google.Api.Gax.Grpc/Rest/PartialDecodingStreamReader.cs

```diff
@@ -222,7 +237,40 @@ private static async Task<T> AwaitWithCancellation<T>(Task<T> task, Cancellation
         cancellationToken.ThrowIfCancellationRequested();
         // If we've got this far, then our real task must have completed:
         // the only way for cancellationTask to complete is if cancellationToken
-        // has been cancelled.
-        return task.Result;
+        // has been cancelled. However, we still need to await it instead
+        // of using .Result so that exceptions are propagated appropriately.
+        return await task.ConfigureAwait(false);
+    }
+
+    public ValueTask DisposeAsync()
+    {
+        Dispose();
+        return default;
+    }
+
+    public void Dispose()
+    {
+        if (_disposed)
+        {
+            return;
+        }
+        _disposed = true;
+
+        // Note: the returned task is deliberately ignored. It's fire-and-forget.
+        _textReaderTask.ContinueWith(DisposeReader, TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
Shouldn't this be awaited on DisposeAsync instead, and then have Dispose call DisposeAsync fire and forgetting?
DisposeAsync callers should be able to await for disposal if they wish to do so?